wirecell/gen/linegen.py

```diff
@@ -185,65 +273,77 @@ def generate_line_track(center, tconf, R_wp):
         p0, p1, tconf.t0, tconf.eperstep, tconf.step_size, tconf.track_speed
     )
 
-    metadata = pack_track_metadata(points, times, dir_wp, dir_glb, tconf, R_wp)
+    dir_wps = np.array([global_direction_to_wp_direction(R_wp, dir_glb) for R_wp in R_wps])
+
+    metadata = pack_track_metadata(points, times, dir_wps, dir_glb, tconf, R_wps)
 
     return (depo_sets, metadata)
 
 def generate_and_save_line_track(
-    center, track_config, phi, path_depo, path_meta
+    center, track_config, phi, path_depo, path_meta, plane_idx=0
 ):
     # pylint: disable=too-many-locals
-    R_wp = plane_yy_to_rotation_matrix(phi)
+    R_wps = np.array([plane_yy_to_rotation_matrix(p) for p in phi])
 
-    (depo_sets, metadata) = generate_line_track(center, track_config, R_wp)
+    (depo_sets, metadata) = generate_line_track(center, track_config, R_wps, plane_idx)
 
     np.savez(path_depo, **depo_sets)
 
     if path_meta is None:
         return
 
     with open(path_meta, "wt", encoding = ENCODING) as f:
```

Comment:
Perhaps, it would be marginally useful to save plane_idx to metadata, as in line 345.